tests/application/cms/test_views.py

```diff
@@ -727,6 +728,91 @@ def test_dept_user_should_be_able_to_edit_shared_page(test_app_client, logged_in
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
+        ),
+        follow_redirects=True,
+    )
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+    assert page.find("input", {"id": "db_version_id"}).attrs["value"] == "1"
+
+    data = multidict_from_measure_version_with_overrides(measure_version, db_version_id="1", title="this is the update")
+    response = test_app_client.post(
+        url_for(
+            "cms.edit_measure_version",
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
+        ),
+        data=data,
+        follow_redirects=True,
+    )
+
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+    assert page.find("input", {"id": "db_version_id"}).attrs["value"] == "2"
+    assert measure_version.title == "this is the update"
+
+
+def test_edit_measure_page_updated_with_latest_db_version_id_when_posting_a_conflicting_update(
+    db_session, test_app_client, logged_in_rdu_user
+):
+    measure_version: MeasureVersion = MeasureVersionFactory(status="DRAFT", title="initial title", version="2.0")
+    assert measure_version.db_version_id == 1
+
+    response = test_app_client.get(
+        url_for(
+            "cms.edit_measure_version",
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
+        ),
+        follow_redirects=True,
+    )
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+    assert page.find("input", {"id": "db_version_id"}).attrs["value"] == "1"
+
+    measure_version.title = "new title"
+    db_session.session.commit()
+    assert measure_version.db_version_id == 2
+
+    data = multidict_from_measure_version_with_overrides(measure_version, db_version_id="1", title="try new title")
+    response = test_app_client.post(
+        url_for(
+            "cms.edit_measure_version",
+            topic_slug=measure_version.measure.subtopic.topic.slug,
+            subtopic_slug=measure_version.measure.subtopic.slug,
+            measure_slug=measure_version.measure.slug,
+            version=measure_version.version,
+        ),
+        data=data,
+        follow_redirects=True,
+    )
+
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode("utf-8"), "html.parser")
+    # assert page.find("input", {"id": "db_version_id"}).attrs["value"] == "2"
```

Comment:
We should be able to assert this reliably I think?  If not then remove comment.

Reply:
Sorry yes, this shouldn't be commented-out.